src/App.tsx

```diff
@@ -1,24 +1,197 @@
+      title,
+      completed: false,
+    };
+
+    try {
+      setIsLoading(true);
+      setTempTodo(newTempTodo);
+
+      await addTodo(title, userId);
+      await fetchedTodos();
+    } catch (error) {
+      addErrorMessage(ErrorType.ADD);
+    } finally {
+      setTempTodo(null);
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleDeleteTodo = useCallback(async (todoId: number) => {
+    try {
+      setProcessingTodos(prevId => ([...prevId, todoId]));
+      await deleteTodo(todoId);
+      await fetchedTodos();
+    } catch (error) {
+      addErrorMessage(ErrorType.DELETE);
+    } finally {
+      setProcessingTodos([]);
+    }
+  }, []);
+
+  const handleClearCompleted = () => {
+    completedTodos.forEach(todo => {
+      handleDeleteTodo(todo.id);
+    });
+  };
+
+  const handleUpdateTodo = useCallback(async (
+    todoId: number,
+    title: string,
+    completed: boolean,
+  ) => {
+    try {
+      setProcessingTodos(prevId => ([...prevId, todoId]));
+      await updateTodo(todoId, title, completed);
+      await fetchedTodos();
+    } catch (error) {
+      addErrorMessage(ErrorType.UPDATE);
+    } finally {
+      setProcessingTodos([]);
+    }
+  }, []);
+
+  const handleToggleAllTodos = () => {
+    if (isAllCompleted) {
+      completedTodos.forEach(todo => {
+        const { id, title, completed } = todo;
+
+        handleUpdateTodo(id, title, !completed);
+      });
+
+      return;
+    }
+
+    todos.forEach(todo => {
+      const { id, title } = todo;
+
+      handleUpdateTodo(id, title, true);
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+
+        <Header
+          userId={USER_ID}
+          onAddTodo={handleAddTodo}
+          onAddErrorMessage={addErrorMessage}
+          isLoading={isLoading}
+          isAllCompleted={isAllCompleted}
+          onToggleAllTodos={handleToggleAllTodos}
+        />
+
+        <TodoList
+          todos={visibleTodos}
+          tempTodo={tempTodo}
+          onDeleteTodo={handleDeleteTodo}
+          processingTodos={processingTodos}
+          onUpdateTodo={handleUpdateTodo}
+        />
+
+        {isTodos && (
+          <Footer
+            activeTodosAmount={activeTodosAmount}
+            onSort={handleSortTodos}
+            sortType={sortType}
+            onClearCompletedTodos={handleClearCompleted}
+            isCompletedTodo={isCompletedTodos}
+          />
+        )}
+      </div>
+
+      {errorType && (
```

Comment:
You have already hidden this component inside by changing className, due to this you don't see smooth animation of the appearing error.